### PR TITLE
Remove a whole bunch of stuff

### PR DESCRIPTION
**Pull request by ttuegel**
This is part of Spring Cleaning 2019. I'm making the pull request early because I don't expect there to be merge conflicts with removing such unused or infrequently used things.

This pull request removes `Kore.MetaML.AST` and `Kore.AST.PureToKore`. Removing the former also requires removing the implicit symbols declared in `Kore.Implicit.ImplicitKore`. Removing the latter requires removing `KorePattern`, which is now the same as `PurePattern`.

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

File: kore/app/exec/Main.hs [hunk 1]
```diff
@@ -430,7 +430,4 @@ mainParseSearchPattern indexedModule patternFileName = do
 makePurePattern
     :: VerifiedKorePattern
     -> CommonStepPattern Object
-makePurePattern pat =
-    case fromKorePattern Object pat of
-        Left err -> error (printError err)
-        Right objPat -> objPat
+makePurePattern = id
```
Comment (marick) [inline kore/app/exec/Main.hs]: Will the spring cleaning include deleting such now-useless definitions? Like this one and....

File: kore/bench/exec/Main.hs [hunk 2]
```diff
@@ -173,9 +173,7 @@ execBenchmark root kFile definitionFile mainModuleName test =
                         , declaredVariables =
                             PatternVerifier.emptyDeclaredVariables
                         }
-            purePattern =
-                either (error . printError) id
-                    (fromKorePattern Object verifiedPattern)
+            purePattern = verifiedPattern
```
Comment (marick) [inline kore/bench/exec/Main.hs]: ... and this one?

File: kore/src/Kore/ASTVerifier/SentenceVerifier.hs [hunk 3]
```diff
@@ -76,11 +78,8 @@ verifyUniqueId existing (UnparameterizedId name location) =
   where
     name' = Text.pack name
 
-definedNamesForSentence :: UnifiedSentence param pat -> [UnparameterizedId]
-definedNamesForSentence =
-    applyUnifiedSentence
-        definedNamesForObjectSentence
-        definedNamesForObjectSentence
+definedNamesForSentence :: Sentence Object param pat -> [UnparameterizedId]
+definedNamesForSentence = definedNamesForObjectSentence
```
Comment (marick) [inline kore/src/Kore/ASTVerifier/SentenceVerifier.hs]: ... and this.

File: kore/src/Kore/ASTVerifier/SentenceVerifier.hs [hunk 4]
```diff
@@ -120,39 +119,32 @@ verifySentence
...
-            builtinVerifiers
-            indexedModule
-            attributesVerification
-        )
-        (verifyObjectSentence
-            builtinVerifiers
-            indexedModule
-            attributesVerification
-        )
+    verifyObjectSentence
```
Comment (marick) [inline kore/src/Kore/ASTVerifier/SentenceVerifier.hs]: ... and this.

File: kore/src/Kore/ASTVerifier/SentenceVerifier.hs [hunk 5]
```diff
@@ -358,23 +350,22 @@ verifyAxiomSentence axiom builtinVerifiers indexedModule =
...
 
 buildDeclaredSortVariables
     :: MetaOrObject level
     => [SortVariable level]
-    -> Either (Error VerifyError) (Set.Set UnifiedSortVariable)
+    -> Either (Error VerifyError) (Set.Set (SortVariable level))
 buildDeclaredSortVariables variables =
-    buildDeclaredUnifiedSortVariables
-        (map asUnified variables)
+    buildDeclaredUnifiedSortVariables variables
```
Comment (marick) [inline kore/src/Kore/ASTVerifier/SentenceVerifier.hs]: .... etc. (won't keep listing them)

File: kore/src/Kore/Implicit/ImplicitSorts.hs [hunk 6]
```diff
@@ -12,124 +12,28 @@ module Kore.Implicit.ImplicitSorts where
...
-( charMetaSort, charListMetaSort
-    , nilCharList, nilCharListA, consCharList, consCharListA
-    , appendCharList, appendCharListA
-    , inCharList, inCharListA
-    , deleteCharList, deleteCharListA
-    , charListAxioms
-    )
-  =
-    defineMetaSort CharSort
+(charMetaSort, charListMetaSort) = defineMetaSort CharSort
```
Comment (virgil-serbanuta) [inline kore/src/Kore/Implicit/ImplicitSorts.hs]: I think that these two are the only sorts that we need right now (and the String alias).

Comment (ttuegel) [reply kore/src/Kore/Implicit/ImplicitSorts.hs]: I think you are right, but some of the tests still use the other meta sorts for now.

**Pull request merged**